datawig/imputer.py

```diff
@@ -311,37 +311,9 @@ def checkpoint(epoch, sym, arg, aux):
         self.train_losses, self.test_losses = train_cb.metrics[metric_name], test_cb.metrics[
             metric_name]
 
-    def __build_module(self, iter_train: ImputerIterDf) -> None:
-
-        # construct the losses
-        predictions, loss = self.__make_loss()
-
-        logger.info("Building output symbols")
-        output_symbols = []
-        for col_enc, output in zip(self.label_encoders, predictions):
-            output_symbols.append(
-                mx.sym.BlockGrad(output, name="pred-{}".format(col_enc.output_column)))
-
-        # Get params to fix assuming we don't want to fine tune
-        fine_tune = any([isinstance(feat, ImageFeaturizer) for feat in self.data_featurizers])
-
-        fixed_params = []
-        if not fine_tune:
-            for name in loss.list_arguments():
-                if "image_featurizer" in name:
-                    fixed_params.append(name)
-        else:
-            fixed_params = None
-
-        self.module = mx.mod.Module(
-            mx.sym.Group([loss] + output_symbols),
-            context=self.ctx,
-            data_names=[name for name, dim in iter_train.provide_data],
-            label_names=[name for name, dim in iter_train.provide_label],
-            fixed_param_names=fixed_params
-        )
-
-        self.module.bind(data_shapes=iter_train.provide_data, label_shapes=iter_train.provide_label)
+    def __build_module(self, iter_train: ImputerIterDf) -> mx.mod.Module:
+        mod = _MXNetModule(self.ctx, self.label_encoders, self.data_featurizers, self.final_fc_hidden_units)
```

Comment:
is ``_MXNetModule.__init__`` always used with immediately calling ``.get`` afterwards? then we could consider passing the iterator directly to ``__init__``?

Reply:
not sure, it can be confusing to create an instance of `_MXNetModule` but return a `mxnet.mod.Module`. What do you think?